src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -703,12 +698,55 @@ private static FormulaValue Substitute(IRContext irContext, FormulaValue[] args)
                 instanceNum = (int)nv.Value;
             }
 
+            // Compute max possible memory this operation may need.
+            var sourceLen = source.Value.Length;
+            var matchLen = match.Value.Length;
+            var replacementLen = replacement.Value.Length;
+                        
+            int maxLenChars;
+            if (instanceNum < 0)
+            {
+                // Replace all instances. 
+                // Maximum possible length of Substitute, convert all the Match to Replacement. 
+                // Unicode, so 2B per character.
```

Comment:
>Unicode, so 2B per character. [](http://example.com/codeflow?start=19&length=29)

Could be 3 byte long

Reply:
True - it depends on the encoding. 

.Net uses a 2-byte unicode encoding:
[https://stackoverflow.com/questions/3787826/internal-representation-of-strings-in-c-sharp](https://stackoverflow.com/questions/3787826/internal-representation-of-strings-in-c-sharp)

Although ultimately, this is just an estimate - I can update the comments.

Reply:
I'll use sizeof(char)